libcontainer/cgroups/ebpf/ebpf.go

```diff
@@ -127,12 +188,35 @@ func LoadAttachCgroupDeviceFilter(insts asm.Instructions, license string, dirFd
 		//       we fail-open on a security feature, which is a bit scary.
 		return nil
 	}
-	// If there was more than one old program, give a warning (since this
-	// really shouldn't happen with runc-managed cgroups) and then detach all
-	// the old programs.
-	if len(oldProgs) > 1 {
-		logrus.Warnf("found more than one filter (%d) attached to a cgroup -- removing extra filters!", len(oldProgs))
-		for _, oldProg := range oldProgs {
+	if !useReplaceProg {
+		// If there was more than one old program, give a warning (since this
+		// really shouldn't happen with runc-managed cgroups) and then detach
+		// all the old programs.
+		infoLevel := logrus.DebugLevel
+		if len(oldProgs) > 1 {
+			// Provide some more detailed debugging info.
+			logrus.Warnf("found more than one filter (%d) attached to a cgroup -- removing extra filters!", len(oldProgs))
+			infoLevel = logrus.InfoLevel
+		}
+		for idx, oldProg := range oldProgs {
+			// Can't use Debugf because then it won't show up during CI.
```

Comment:
CI should run runc with --debug?

Reply:
Yes, but the unit tests don't do `logrus.SetLevel(logrus.DebugLevel)` AFAIK -- this is needed so I can debug #2976.

Reply:
Alternatively we can add `logrus.SetLevel(logrus.DebugLevel)` too all test files. That might not be a bad idea actually...